sdk/spring/spring-integration-azure-core/src/main/java/com/azure/spring/integration/core/handler/DefaultMessageHandler.java

```diff
@@ -87,7 +88,7 @@ protected void handleMessageInternal(Message<?> message) {
 
         final Mono<Void> mono = this.sendOperation.sendAsync(dest, messageToSend);
 
-        if (this.sync) {
+        if (this.sync && !Schedulers.isInNonBlockingThread()) {
```

Comment:
Please add comment here, to explain in which scenario will this happen, better to add some code example for it: